java/src/main/java/org/hyperledger/fabric/client/Checkpoint.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2022 IBM All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.fabric.client;
+
+/**
+ * Checkpoint provides the current position for event processing.
+ */
+public interface Checkpoint {
+    /**
+     *
+     * @return  BlockNumber in which the next event is expected.
```

Comment:
Needs some actual description.
```suggestion
     * The block number in which the next event is expected.
     * @return A ledger block number.
```